dask_expr/_expr.py

```diff
@@ -407,6 +407,10 @@ def columns(self) -> list:
         except Exception:
             raise
 
+    @functools.cached_property
+    def unique_partition_mapping_columns(self):
+        return set()
```

Comment:
Very high level, I think the name doesn't make any sense any longer. I started this with this name encoding a different information.

I would also like to see a doc string here describing this property because this is highly nontrivial, I think.

What I initially planned is this

```python
    @functools.cached_property
    def unique_partition_mapping_columns(self):
        """A set of tuples of columns that identify a unique, bijective mapping from values to partition

        Example::

            unique_partition_mapping_columns = {
                ("A", "B"),
                "C",
            }

            Partition 1
            | A | B | C | D |
            |---|---|---|---|
            | 1 | 1 | 1 | 1 |
            | 1 | 2 | 1 | 2 |
            | 1 | 3 | 1 | 3 |

            Partition 2
            | A | B | C | D |
            |---|---|---|---|
            | 1 | 4 | 2 | 1 |
            | 1 | 5 | 3 | 2 |
            | 2 | 3 | 4 | 3 |
        """
        return set()
```

but this is not sufficient to remove unnecessary merges. I think there is some implicit information encoded in here that we should make explicit.

Is it...

- All columns that satisfy the above property and were created by a shuffle?

or is it something else? Because if it's the shuffle thing, I don't think this can be a set but there is always exactly one tuple (we know of) that satisfies this condition

Reply:
> is it something else? Because if it's the shuffle thing, I don't think this can be a set but there is always exactly one tuple (we know of) that satisfies this condition

It is a shuffle, but also a merge based shuffle. And merge based shuffle have 2 tuples, the columns that were shuffled on on the left and the corresponding columns on the right. Plain shuffles will always be 1 tuple, that's correct.

Reply:
ok, that makes sense.

In this case I would name this property to something like `unique_column_mapping_from_shuffle`